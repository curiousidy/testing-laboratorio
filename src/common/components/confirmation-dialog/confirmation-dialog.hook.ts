import React from 'react';
import { createEmptyLookup, Lookup } from 'common/models';

export const useConfirmationDialog = () => {
  const [isOpen, setIsOpen] = React.useState(false);
  const [itemToDelete, setItemToDelete] = React.useState(createEmptyLookup());

  return {
    isOpen,
    itemToDelete,
    onAccept: () => {
      setItemToDelete(createEmptyLookup());
    },
    onClose: () => setIsOpen(false),
    onOpenDialog: (item: Lookup) => {
      setIsOpen(item ? true : false);
      setItemToDelete(item ? item : createEmptyLookup());
    },
  };
};
