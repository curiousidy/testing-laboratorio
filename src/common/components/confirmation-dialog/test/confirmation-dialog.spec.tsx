import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ConfirmationDialogComponent } from '../confirmation-dialog.component';

describe('Confirmation Dialog Component', () => {
    test('verifies that the title, message and buttons are rendered, and that the onAccept and onClose functions are called when the buttons are clicked.', () => {
        //Arrange
        const isOpen = true;
        const onAccept = jest.fn();
        const onClose = jest.fn();
        const title = 'Diálogo de confirmación';
        const labels = {
            closeButton: 'Cerrar',
            acceptButton: 'Aceptar',
        };
        const children = <div>Mensaje de confirmación</div>;

        //Act
        const { getByText } = render(<ConfirmationDialogComponent
            isOpen={isOpen}
            onAccept={onAccept}
            onClose={onClose}
            title={title}
            labels={labels}
            children={children}
        />)

        //Assert
        expect(getByText(title)).toBeInTheDocument();
        expect(getByText('Mensaje de confirmación')).toBeInTheDocument();

        fireEvent.click(getByText('Cerrar'));
        expect(onClose).toHaveBeenCalled();

        fireEvent.click(getByText('Aceptar'));
        expect(onAccept).toHaveBeenCalled();
        
    })

    test('snapshoot component', () => {
        //Arrange
        const isOpen = true;
        const onAccept = jest.fn();
        const onClose = jest.fn();
        const title = 'Diálogo de confirmación';
        const labels = {
            closeButton: 'Cerrar',
            acceptButton: 'Aceptar',
        };
        const children = <div>Mensaje de confirmación</div>;

        //Act
        const { baseElement } = render(<ConfirmationDialogComponent
            isOpen={isOpen}
            onAccept={onAccept}
            onClose={onClose}
            title={title}
            labels={labels}
            children={children}
        />)
       
        //Assert
       expect(baseElement).toMatchSnapshot();
        
    })
})
