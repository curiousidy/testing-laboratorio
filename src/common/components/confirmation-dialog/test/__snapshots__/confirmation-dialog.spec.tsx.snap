// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Dialog Component snapshoot component 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r1:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r1:"
        >
          Diálogo de confirmación
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div>
            Mensaje de confirmación
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Cerrar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Aceptar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
